libtransmission/port-forwarding-natpmp.cc

```diff
@@ -96,7 +96,7 @@ tr_natpmp::PulseResult tr_natpmp::pulse(tr_port local_port, bool is_enabled)
         }
     }
 
-    if ((state_ == State::Idle || state_ == State::Err) && is_mapped_ && (!is_enabled || local_port_ != local_port))
+    if (state_ <= State::Err && is_mapped_ && (!is_enabled || local_port_ != local_port))
```

Comment:
For maintainability, enumerations shouldn't have ordered comparisons (since new enumeration values could be inserted or re-ordered, causing side-effects). Also that makes it harder to read/review for humans, since we don't know the affected enumeration values without looking them up.

I propose you revert that line:
```suggestion
    if ((state_ == State::Idle || state_ == State::Err) && is_mapped_ && (!is_enabled || local_port_ != local_port))
```

Reply:
6612dd5